astropy/io/ascii/ui.py

```diff
@@ -872,8 +872,26 @@ def write(table, output=None, format=None, Writer=None, fast_writer=True, *,
     if output is None:
         output = sys.stdout
 
-    table_cls = table.__class__ if isinstance(table, Table) else Table
-    table = table_cls(table, names=kwargs.get('names'))
+    # Ensure that `table` is a Table subclass.
+    names = kwargs.get('names')
+    if isinstance(table, Table):
```

Comment:
This has been refactored a bit but the only real diff is setting `serialized_meta` where appropriate.